pkg/motor/login.go

```diff
@@ -40,15 +43,38 @@ func (mtr *motorNodeImpl) Login(request mt.LoginRequest) (mt.LoginResponse, erro
 
 // LoginWithKeys creates a motor node from a LoginRequest
 func (mtr *motorNodeImpl) LoginWithKeys(request mt.LoginWithKeysRequest) (mt.LoginResponse, error) {
-	if request.Did == "" {
+	if request.AccountId == "" {
 		return mt.LoginResponse{}, fmt.Errorf("did must be provided")
 	}
 
-	mtr.Address = request.Did
+	// Create Client instance
+	mtr.Cosmos = client.NewClient(client.ConnEndpointType_BETA)
+
+	// if the given ID is an alias, first fetch the address
+	var whoIs *types.WhoIs
+	if strings.HasSuffix(request.AccountId, ".snr") {
+		whoIsResp, err := mtr.QueryWhoIsByAlias(mt.QueryWhoIsByAliasRequest{
+			Alias: request.AccountId,
+		})
+		if err != nil {
+			return mt.LoginResponse{}, fmt.Errorf("query WhoIs by alias: %s", err)
+		}
+		whoIs = whoIsResp.WhoIs
+	} else {
+		whoIsResp, err := mtr.QueryWhoIs(mt.QueryWhoIsRequest{
+			Did: request.AccountId,
+		})
+		if err != nil {
+			return mt.LoginResponse{}, fmt.Errorf("query WhoIs: %s", err)
+		}
+		whoIs = whoIsResp.WhoIs
+	}
```

Comment:
String parsing is currently what's being done in Flutter so I'm copying that logic here.

I think having it as one field simplifies the interface for the developer, because in most cases this value is going to come from a single text field or something.

Reply:
I saw where you’re coming from josh and I think I agree with Brayden. Otherwise what happens if you pass conflicting information?